src/js/_enqueues/admin/site-icon.js

```diff
@@ -1,138 +1,180 @@
 				$( '#site_icon_hidden_field' ).val(attachment.id);
-				switchToUpdate(attachment.url);
+				switchToUpdate(attachment.attributes.url);
 				frame.close();
-				// Start over with a frame that is so fresh and so clean clean.
-				frame = null;
-			});
-
-			// When an image is selected, run a callback.
-			frame.on( 'select', function() {
-				// Grab the selected attachment.
-				var attachment = frame.state().get('selection').first();
-
-				if ( attachment.attributes.height === $el.data('size') && $el.data('size') === attachment.attributes.width ) {
-					// Set the value of the hidden input to the attachment id.
-					$( '#site_icon_hidden_field').val(attachment.id);
-					switchToUpdate(attachment.attributes.url);
-					frame.close();
-				} else {
-					frame.setState( 'cropper' );
-				}
-			});
-
-			frame.open();
+			} else {
+				frame.setState( 'cropper' );
+			}
 		});
-	});
 
-	function switchToUpdate( url ){
-		// Set site-icon-img src to the url and remove the hidden class.
-		$( '#site-icon-preview').find('img').not('.browser-preview').each( function(i, img ){
-			$(img).attr('src', url );
+		frame.open();
+	});
+})( jQuery );
+
+/**
+ * Updates the site icon preview with the specified URL.
+ *
+ * @since 6.5.0
+ *
+ * @param {string} url The URL of the site icon image.
+ */
+function switchToUpdate( url ) {
+	var chooseButton = jQuery( '#choose-from-library-button' );
+	var iconPreview = jQuery( '#site-icon-preview' );
+	// Set site-icon-img src to the url and remove the hidden class.
+	iconPreview.find( 'img' ).not( '.browser-preview' )
+		.each( function( i, img ) {
+			jQuery(img).attr( 'src', url );
+		});
+	iconPreview.removeClass( 'hidden' );
+	// Remove hidden class from remove.
+	jQuery( '#js-remove-site-icon' ).removeClass( 'hidden' );
+	// If the button is not in the update state, swap the classes.
+	if ( chooseButton.attr( 'data-state' ) !== '1' ) {
+		chooseButton.attr({
+			'class': chooseButton.attr( 'data-alt-classes' ),
+			'data-alt-classes': chooseButton.attr( 'class' ),
+			'data-state': '1'
 		});
-		$( '#site-icon-preview' ).removeClass( 'hidden' );
-		// Remove hidden class from remove.
-		$( '#js-remove-site-icon' ).removeClass( 'hidden' );
-		// If the button is not in the update state, swap the classes.
-		if( $( '#choose-from-library-link' ).attr( 'data-state' ) !== '1' ){
-			var classes = $( '#choose-from-library-link' ).attr( 'class' );
-			$( '#choose-from-library-link' ).attr( 'class', $( '#choose-from-library-link' ).attr('data-alt-classes') );
-			$( '#choose-from-library-link' ).attr( 'data-alt-classes', classes );
-			$( '#choose-from-library-link' ).attr( 'data-state', '1' );
-		}
-
-		// swap the text of the button
-		$( '#choose-from-library-link' ).text( $( '#choose-from-library-link' ).attr( 'data-update-text' ) );
 	}
 
+	// Swap the text of the button.
+	chooseButton.text(
+		chooseButton.attr( 'data-update-text' )
+	);
+	chooseButton.trigger( 'focus' );
```

Comment:
This should not be necessary because the media modal dialog already sets focus back to the button.